src/main/java/io/jenkins/plugins/orka/AgentTemplate.java

```diff
@@ -147,7 +146,12 @@ public OrkaVerificationStrategy getVerificationStrategy() {
     }
 
     public OrkaProvisionedAgent provision() throws IOException, FormException {
-        this.ensureConfigurationExist();
+        ConfigurationResponse configurationResponse = this.ensureConfigurationExist();
+        if (configurationResponse != null && !configurationResponse.isSuccessful()) {
+            logger.warning("Creating VM configuration failed with: " + Utils.getErrorMessage(configurationResponse));
```

Comment:
To me it's a bit confusing to create the configuration in one function and check the errors from this operation here (in another function). I would suggest to move the error checking in the `ensureConfigurationExist` method and return `true` or `false` in there based on the result. Then check the result here and return `null` in case of not existing configuration.